doc/california_law/California_law.md

```diff
@@ -0,0 +1,236 @@
+# Default Credential Management for California Law SB-327 Conformance #
+
+##  1. <a name='TableofContent'></a>Table of Content
+
+<!-- vscode-markdown-toc -->
+* 1. [Table of Content](#TableofContent)
+	* 1.1. [Revision](#Revision)
+	* 1.2. [Scope](#Scope)
+	* 1.3. [Definitions/Abbreviations](#DefinitionsAbbreviations)
+	* 1.4. [Overview](#Overview)
+	* 1.5. [Requirements](#Requirements)
+	* 1.6. [Architecture Design](#ArchitectureDesign)
+	* 1.7. [High-Level Design](#High-LevelDesign)
+		* 1.7.1. [Flow description](#Flowdescription)
+		* 1.7.2. [California Law Constrains](#CaliforniaLawConstrains)
+	* 1.8. [Init Flow](#InitFlow)
+		* 1.8.1. [Compilation](#Compilation)
+		* 1.8.2. [Dependencies](#Dependencies)
+		* 1.8.3. [Feature default](#Featuredefault)
+	* 1.9. [SAI API](#SAIAPI)
+	* 1.10. [Configuration and management](#Configurationandmanagement)
+	* 1.11. [Warmboot and Fastboot Design Impact](#WarmbootandFastbootDesignImpact)
+	* 1.12. [Restrictions/Limitations](#RestrictionsLimitations)
+	* 1.13. [Upgrade Flow](#UpgradeFlow)
+	* 1.14. [Test Plan](#TestPlan)
+		* 1.14.1. [Unit Test cases](#UnitTestcases)
+		* 1.14.2. [System Test cases](#SystemTestcases)
+		* 1.14.3. [Pasword Change Flow](#PasswordChangeFlow)
+	* 1.15. [3rd Party Components](#rdPartyComponents)
+		* 1.15.1. [PW Force Expiration](#WForceExpire)
+		* 1.15.2. [Pam Unix](#PAMUNIX)
+
+
+<!-- vscode-markdown-toc-config
+	numbering=true
+	autoSave=true
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+
+
+###  1.1. <a name='Revision'></a>Revision
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 08/2022 |   Andriy Dobush  | Phase 1 Design     |
+
+###  1.2. <a name='Scope'></a>Scope
+
+This California Law hld doc described the requirements, architecture and configuration details of California Law feature in switches SONIC OS based.
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations
+
+	PW - password
+	PASSWH - Password Hardening
+	PAM - Pluggable Authentication Modules
+	CL - Default Credential Management for California Law SB-327 Conformance
+
+###  1.4. <a name='Overview'></a>Overview
+
+Most users won't change the password because of various reasons: forgot to change, don't realize default password should be changed and other. As a result
+default passwords in connected devices is one of a primary attack method used by malicious actors. The main motivation and functionality of the feature is to
+force user to change default password on first login.
+
+###  1.5. <a name='Requirements'></a>Requirements
+
+ - Force change of default password on first login
+ - Support several default users
```

Comment:
Suggest clearify 'default user', as my understand we don't need expire users who's shell is /bin/false or /usr/sbin/nologin.